src/Formatter/Formatter.php

```diff
@@ -20,6 +20,27 @@
 
 class Formatter
 {
+    protected $configurationCallbacks = [];
+
+    protected $parsingCallbacks = [];
+
+    protected $renderingCallbacks = [];
+
+    public function addConfigurationCallback($callback)
+    {
+        $this->configurationCallbacks[] = $callback;
+    }
+
+    public function addParsingCallback($callback)
+    {
+        $this->parsingCallbacks[] = $callback;
+    }
+
+    public function addRenderingCallback($callback)
+    {
+        $this->renderingCallbacks[] = $callback;
+    }
```

Comment:
Please move these methods down to the other methods. :pray:

Reply:
Done!